src/gl/lineFeature.js

```diff
@@ -151,6 +152,19 @@ var gl_lineFeature = function (arg) {
         position.push(pos.x);
         position.push(pos.y);
         position.push(pos.z || 0.0);
+        if (!j) {
+          firstpos = pos;
+        }
+      }
+      if (lineItem.length > 2 && closedFunc(data[i], i)) {
```

Comment:
For sub-components, like points within a line, the style information is obtained by calling a function with four parameters: the current sub-item (a vertex, for instance), the sub-item's index, the current item (the line), and the item index.